### PR TITLE
add plain-text-with-link component

### PR DESCRIPTION
**Pull request by boaz0**
Cost management needs to include a link to the documentation in the plain text. However,1) there is no link component in `componentMapper` and  2) the link needs to be in the same line of the description text.

In this commit, I created a new component that takes `label` (the description text) and link data (url and text) to generate this kind of pattern.

screenshot to illustrate the new component.
image

//cc @rvsia @karelhala

### CODE REVIEW

File: src/PresentationalComponents/PlainTextWithLink.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Text, TextVariants } from '@patternfly/react-core';
+
+export const PlainTextWithLink = ({ label, linkText, linkUrl }) => {
+    return (
+        <Text component={ TextVariants.p }>
+            {label}
+            &nbsp;&nbsp;
```
Comment (rvsia) [inline src/PresentationalComponents/PlainTextWithLink.js:9]: You can put a left marging to the link. i.e. `pf-u-ml-md`

File: src/PresentationalComponents/PlainTextWithLink.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+            <Text component={ TextVariants.a } href={ linkUrl }>{linkText}</Text>
```
Comment (rvsia) [inline src/PresentationalComponents/PlainTextWithLink.js:10]: What about to open links in a new tab?

File: src/PresentationalComponents/PlainTextWithLink.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+        </Text>
+    );
+};
+
+PlainTextWithLink.propTypes = {
+    label: PropTypes.string,
+    linkText: PropTypes.string,
```
Comment (rvsia) [inline src/PresentationalComponents/PlainTextWithLink.js:17]: I think these props should be required.

Comment (boaz0) [reply src/PresentationalComponents/PlainTextWithLink.js]: Thanks! I added `isRequired`.

File: src/PresentationalComponents/PlainTextWithLink.js (hunk 2 above)
Comment (boaz0) [reply src/PresentationalComponents/PlainTextWithLink.js]: Yep, I missed that part. Thanks

File: src/PresentationalComponents/PlainTextWithLink.js (hunk 1 above)
Comment (boaz0) [reply src/PresentationalComponents/PlainTextWithLink.js]: Cool! Thanks

File: src/PresentationalComponents/PlainTextWithLink.js [hunk 4]
```diff
@@ -0,0 +1,18 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Text, TextVariants } from '@patternfly/react-core';
+
+export const PlainTextWithLink = ({ label, linkText, linkUrl }) => {
+    return (
+        <Text component={ TextVariants.p }>
+            {label}
+            <Text className="pf-u-ml-md" component={ TextVariants.a } target="_blank" href={ linkUrl }>{linkText}</Text>
```
Comment (rvsia) [inline src/PresentationalComponents/PlainTextWithLink.js:9]: I would add `rel="noreferrer noopener"` for better security.

Comment (boaz0) [reply src/PresentationalComponents/PlainTextWithLink.js:9]: Oh thanks! I forgot about that!

File: src/PresentationalComponents/PlainTextWithLink.js [hunk 5]
```diff
@@ -0,0 +1,18 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Text, TextVariants } from '@patternfly/react-core';
+
+export const PlainTextWithLink = ({ label, linkText, linkUrl }) => {
+    return (
+        <Text component={ TextVariants.p }>
```
Comment (Hyperkid123) [inline src/PresentationalComponents/PlainTextWithLink.js:7]: The way this is structured is probably not flexible enough. What, if there are 3 words and the link, should be middle one?

Can we come up with some template string-like syntax? For instance if we check the react-intl format message API we could endup with something like this:

```jsx
// input
Foo <link>bar</link> bazz
// output
<Text {...someProps}>Foo <Text component="a" {...moreProps}>bar</Text> bazz</Text>
```
It should be easy enough to replace the rich text tag with react component using string operations

Comment (rvsia) [reply src/PresentationalComponents/PlainTextWithLink.js:7]: As I proposed some time ago, we could go fully markdown way as we need store the schema in the API. But maybe it's too complex to just show a text with a link.

Comment (Hyperkid123) [reply src/PresentationalComponents/PlainTextWithLink.js:7]: I think this is a common enough component that will require fairly powerful customization later on. I ain't a huge fan of adding MD dependencies just for a simple link component but I can see the appeal. We could "steal" the MD way of defining the link though. One way or another I think we need a bit more flexible definition that this one.

Comment (boaz0) [reply src/PresentationalComponents/PlainTextWithLink.js:7]: @Hyperkid123 - I am not sure how can we replace "<link>" with "<Text component...>" mind dropping some refs?
Thanks.

Comment (Hyperkid123) [reply src/PresentationalComponents/PlainTextWithLink.js:7]: @boaz0 you could use regexp for instance. You can find the non-link string like this: 
```js
"Foo <link>bar</link> bazz".split(/<link>.*<\/link>/) // output => ["Foo ", " bazz"]
```

And then extract the text inside the link tag and create a react link component

```js
const x = "Foo <link>bar</link> bazz"
x.slice(x.search(/<link>/) + 6, x.search(/<\/link>/)) // output => bar
```
Now you have all 3 fragments and you can compose the component.

Comment (boaz0) [reply src/PresentationalComponents/PlainTextWithLink.js:7]: IMHO, sounds a little bit odd and not secure (DDoS attack, idk). But if that is what you want, then I guess we can do that.

Comment (Hyperkid123) [reply src/PresentationalComponents/PlainTextWithLink.js:7]: What do you mean by DDos atack?

Comment (boaz0) [reply src/PresentationalComponents/PlainTextWithLink.js:7]: Let me rephrase myself. I do **not** think this is a bad idea. I am just concerned that it is done in run-time versus in build time. If you are OK with this, I don't mind. I also was hoping there was a function/method/component that we can use in React Intl that can do it for us but I don't care doing it.

Comment (Hyperkid123) [reply src/PresentationalComponents/PlainTextWithLink.js:7]: Right. The runtime risk is minimal IMO. This would do basically the same thing as your initial implements but is slightly more flexible. You can definitely use react intl if you want. It has rich text API with exactly the same syntax. And obviously their implementation is much more sophisticated.

**Comment by boaz0** [comment]
@Hyperkid123 can you review this?

**Comment by boaz0** [comment]
Thank ya'all
